File: components/support/nimbus-fml/src/backends/swift/templates/EnumTemplate.swift
{% import "macros.swift" as swift %}
{% let inner = self.inner() %}
{% let class_name = inner.name()|class_name %}
{{ inner.doc()|comment("") }}
public enum {{ class_name }}: String, CaseIterable, Codable {
    {% for variant in inner.variants() %}
    {{ variant.doc()|comment("    ") }}
    case {{ variant.name()|enum_variant_name }} = {{variant.name()|quoted}}
    {% endfor %}

    public static func enumValue(_ s: String?) -> {{class_name}}? {
        guard let s = s else {
            return nil
        }
        return {{class_name}}(rawValue: s)
    }
}

File: components/support/nimbus-fml/src/backends/kotlin/templates/EnumTemplate.kt
/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/. */

{% import "macros.kt" as kt %}
{% let inner = self.inner() %}
{% let class_name = inner.name()|class_name %}

{{ inner.doc()|comment("") }}
enum class {{class_name}} {
    {% for v in inner.variants() %}
    {{ v.doc()|comment("    ") }}
    {{ v.name()|enum_variant_name }}{% if !loop.last %},{% endif %}{% endfor %};

    companion object {
        private val enumMap: Map<String, {{class_name}}> by lazy {
            mapOf({% for v in inner.variants() %}
                {{ v.name()|quoted }} to {{class_name}}.{{ v.name()|enum_variant_name }}{% if !loop.last %},{% endif %}{% endfor %})
        }

        fun enumValue(string: String): {{class_name}}? = enumMap[string]
    }

    fun toJSONString() =
        when (this) {
            {%- for v in inner.variants() %}
            {{ v.name()|enum_variant_name }} -> {{ v.name()|quoted }}{% endfor %}
        }
}
